src/python/roms/__init__.py

```diff
@@ -1,141 +1,90 @@
-    """Return the ROM directory, i.e., a list of all supported ROMs."""
-    md5s = resources.files(__name__).joinpath("md5.txt")
-    if not md5s.exists():
-        raise FileNotFoundError(
-            "ROM md5 resource couldn't be found. "
-            "Are you running from a development environment? "
-        )
-    with md5s.open() as fp:
-        lines = filter(
-            lambda line: line.strip() and not line.startswith("#"), fp.readlines()
-        )
-        roms = [pathlib.Path(rom) for _, rom in map(str.split, lines)]
-        return [utils.rom_id_to_name(rom.stem) for rom in roms]
-
-
-@functools.lru_cache(maxsize=None)
-def __getattr__(name: str) -> pathlib.Path:
-    """Return the path to a ROM."""
-    roms = __dir__()
-    if name not in roms:
-        raise AttributeError(f"No ROM named {name}. Supported ROMs: {', '.join(roms)}")
-
-    path = _resolve_rom(name)
-    if path is None:
-        raise AttributeError(
-            f"Failed to resolve ROM `{name}` from plugins "
-            f"{', '.join(map(lambda plugin: f'`{repr(plugin)}`', _ROM_PLUGIN_REGISTRY))}. "
-            "If you own a license to use the necessary ROMs for research purposes you can download them "
-            f"via `pip install autorom[accept-rom-license]`. Otherwise, you should try importing `{name}` "
-            f"via the command `ale-import-roms`. If you believe this is a mistake perhaps your copy of `{name}` "
-            "is unsupported. To check if this is the case try providing the environment variable "
-            "`PYTHONWARNINGS=default::ImportWarning:ale_py.roms`. For more information see: "
-            "https://github.com/mgbellemare/Arcade-Learning-Environment#rom-management"
-        )
-    return path
-
-
-def register_plugin(plugin: plugins.Plugin, *, index: int = 0) -> None:
-    """Register a ROM plugin."""
-    if not issubclass(type(plugin), plugins.Plugin):
-        raise ValueError(f"{repr(plugin)} is not a valid ROM plugin.")
-    _ROM_PLUGIN_REGISTRY.insert(index, plugin)
-    __getattr__.cache_clear()
-
-
-__all__ = ["register_plugin"] + __dir__()
+        # grab the rom name from the member name, ie: `pong.bin`
+        # member.name is ROM/rom_name/rom_name.bin, so we just want the last thing
+        rom_name = member.name.split("/")[-1]
+
+        # assert that this member.name should be supported
+        assert (
+            rom_name in all_roms
+        ), f"File {rom_name} not supported. Please report this to a dev."
+
+        # extract the rom file from the archive
+        rom_bytes = tar_fp.extractfile(
+            member
+        ).read()  # pyright: ignore[reportOptionalMemberAccess]
+
+        # get hash
+        md5 = hashlib.md5()
+        md5.update(rom_bytes)
+        md5_hash = md5.hexdigest()
+
+        # assert that the hash matches
+        assert (
+            md5_hash == all_roms[rom_name]
+        ), f"Rom {rom_name}'s hash does not match what was expected. Please report this to a dev."
```

Comment:
Do you want to include the expected and actual hashes for debugging?

Reply:
Probably not, I doubt that's very useful